packages/react-components/react-search/library/src/components/SearchBox/useSearchBox.tsx

```diff
@@ -46,13 +46,16 @@ export const useSearchBox_unstable = (props: SearchBoxProps, ref: React.Ref<HTML
   // Tracks the focus of the component for the contentAfter and dismiss button
   const [focused, setFocused] = React.useState(false);
 
-  const onFocus = useEventCallback(() => {
+  const onFocus = React.useCallback(() => {
     setFocused(true);
-  });
+  }, [setFocused]);
 
-  const onBlur: React.FocusEventHandler<HTMLSpanElement> = useEventCallback(ev => {
```

Comment:
**question:** wondering what's the real motivation and perf benefits of using this abstraction in our codebase ? cc @ling1726 

when even react docs explicitly mention another solution should be used ⬇️

https://legacy.reactjs.org/docs/hooks-faq.html#how-to-avoid-passing-callbacks-down

Reply:
@Hotell the answer is next item in FAQ (see "How to read an often-changing value from useCallback?"). We don't know what underlaying components will be (people can recompose components), so we are _trying_ to avoid passing unstable references down (at the same time this pattern is documented nowhere 🥲).